src/Spice86.Core/Emulator/OperatingSystem/Structures/DosMemoryControlBlock.cs

```diff
@@ -79,11 +79,18 @@ public class DosMemoryControlBlock : MemoryBasedDataStructureWithBaseAddress {
     public bool IsValid => IsLast || IsNonLast;
 
     /// <summary>
-    /// Returns the next MCB in the MCB chain.
+    /// Returns the next MCB in the next MCB in chain, along with a boolean status.
     /// </summary>
-    /// <returns></returns>
-    public DosMemoryControlBlock Next() {
-        return new DosMemoryControlBlock(Memory, BaseAddress + MemoryUtils.ToPhysicalAddress((ushort)(Size + 1), 0));
+    /// <param name="next">The output variable containing the next MCB, or this MCB.</param>
```

Comment:
Why not keeping the same semantic (next) but returning null when the end of memory is reached?